gen/gen.go

```diff
@@ -414,31 +414,37 @@ import (
 	"github.com/swaggo/swag"
 )
 
-var doc = ` + "`{{ printDoc .Doc}}`" + `
+var doc_{{ .InstanceName }} = ` + "`{{ printDoc .Doc}}`" + `
 
-type swaggerInfo struct {
+type swaggerInfo_{{ .InstanceName }} struct {
 	Version     string
 	Host        string
 	BasePath    string
 	Schemes     []string
 	Title       string
 	Description string
+	instanceName string
 }
 
-// SwaggerInfo holds exported Swagger Info so clients can modify it
-var SwaggerInfo = swaggerInfo{
+func (i *swaggerInfo_{{ .InstanceName }}) InstanceName() string {
+	return i.instanceName
+}
+
+// SwaggerInfo_{{ .InstanceName }} holds exported Swagger Info so clients can modify it
+var SwaggerInfo_{{ .InstanceName }} = swaggerInfo_{{ .InstanceName }}{
 	Version:     {{ printf "%q" .Version}},
 	Host:        {{ printf "%q" .Host}},
 	BasePath:    {{ printf "%q" .BasePath}},
 	Schemes:     []string{ {{ range $index, $schema := .Schemes}}{{if gt $index 0}},{{end}}{{printf "%q" $schema}}{{end}} },
 	Title:       {{ printf "%q" .Title}},
 	Description: {{ printf "%q" .Description}},
+	instanceName: {{ printf "%q" .InstanceName }},
 }
 
-type s struct{}
+type s_{{ .InstanceName }} struct{}
 
-func (s *s) ReadDoc() string {
-	sInfo := SwaggerInfo
+func (s *s_{{ .InstanceName }}) ReadDoc() string {
+	sInfo := SwaggerInfo_{{ .InstanceName }}
```

Comment:
Why do we need an additional s_InstanceName struct when we can define the ReadDoc() for swaggerInfo_InstanceName?
```
func (i *swaggerInfo_{{ .InstanceName }}) ReadDoc() string { 
```

Reply:
How are looking on defining swaggerInfo struct inside swag as exported struct and generating only var SwaggerInfo_InstanceName with all methods?

Reply:
replace
```
func (s *s_{{ .InstanceName }}) ReadDoc() string {
```

with 
```
func (s *swaggerInfo_{{ .InstanceName }}) ReadDoc() string { 
```

remove
```
type s_{{ .InstanceName }} struct{}
sInfo := SwaggerInfo_{{ .InstanceName }}
```

update 
```
	if err := t.Execute(&tpl, sInfo); err != nil {
```
to 
```
	if err := t.Execute(&tpl, s); err != nil {
```

Reply:
Does that make sense now ?

Reply:
How about doing it this way, it wont break any current cases

Reply:
Removing ReadDoc() function will introduce breaking changes. 

See yourself by regenerating the documentation for example/basic and building the project with the proposed way.

Reply:
I've tried to, with importing local Swagger info struct, worked fine

Reply:
I'm referring to swag.SwaggerInfo, which matches Swagger interface

Reply:
@ubogdan what do you think about changes?

Reply:
@Ferrany1 congratulations! moving the code from string constant to swag enable us to properly write unit tests against it. 
I'm quite caught with different things, but I will try to do a CR soon.